typing/typecore.ml

```diff
@@ -9549,12 +9555,27 @@ and type_let_def_wrap_warnings
                    event *)
                 let name = Ident.name id in
                 let used = ref false in
+                let mutable_ =
+                  (match vd.val_kind with Val_mut _ -> true | _ -> false)
+                in
+                let unused_mutable = ref mutable_ in
                 if not (name = "" || name.[0] = '_' || name.[0] = '#') then
                   add_delayed_check
                     (fun () ->
-                      if not !used then
+                      if not !used || !unused_mutable then
```

Comment:
I think the below match is pretty hard to understand, as a reader.  I suggest the following changes:
* `unused_mutable`'s meaning is a bit unclear - does it mean "not used" or "not mutated".  Suggest changing it to "unmutated" or similar.
* Maybe instead of having the big pattern match, you can have an if then else like:
```
if not !used then
  ...
else if !unmutated then
  ...
;
```

And have comments in each case explaining it.